usaspending_api/common/mixins.py

```diff
@@ -155,21 +155,24 @@ def filter_records(self, request, *args, **kwargs):
         # process to accept a list of paramaters
         # and create filters without needing to know about the structure
         # of the request itself.
+        filters = None
+        filter_map = kwargs.get('filter_map', {})
+        fg = FilterGenerator(queryset.model, filter_map=filter_map)
+
         if len(request.data):
             fg = FilterGenerator(queryset.model)
             filters = fg.create_from_request_body(request.data)
-            return queryset.filter(filters).distinct()
         else:
-            filter_map = kwargs.get('filter_map', {})
-            fg = FilterGenerator(queryset.model, filter_map=filter_map)
-            filters = fg.create_from_query_params(request.query_params)
-            # add fiscal year to filters if requested
-            # deprecated: we plan to start storing fiscal years in the database
-            if request.query_params.get('fy'):
-                fy = FiscalYear(request.query_params.get('fy'))
-                fy_arguments = fy.get_filter_object('date_signed', as_dict=True)
-                filters = {**filters, **fy_arguments}
-            return queryset.filter(**filters).distinct()
+            filters = Q(**fg.create_from_query_params(request.query_params))
+
+        # Handle FTS vectors
+        if len(fg.search_vectors) > 0:
```

Comment:
Now I'm really curious if 169-172 could be replaced by `vector_sum = sum(fg.search_vectors)`.  If search vectors support `+=` I bet they support `sum`.